os/fs/driver/mtd/smart.c

```diff
@@ -3348,6 +3348,107 @@ static int smart_relocate_block(FAR struct smart_struct_s *dev, uint16_t block)
 	return ret;
 }
 
+int find_free_physical_sector(uint16_t allocblock, FAR struct smart_struct_s *dev, uint8_t *sector_buff, uint16_t *physicalsector, bool *bitflipped)
```

Comment:
It doesn't find sector, just check validation of block. name is not acceptable, in my opinion.